Source/FieldSolver/ElectrostaticSolver.cpp

```diff
@@ -52,22 +53,62 @@
 
 using namespace amrex;
 
+void WarpX::ResetRho_fp ()
+{
+    // Reset the charge density
+    for (int lev = 0; lev <= max_level; lev++) {
+        rho_fp[lev]->setVal(0.);
+    }
+}
+
+void WarpX::ResetFields ()
+{
+    // Reset the Efield and Bfield
+    for (int lev = 0; lev <= max_level; lev++) {
+        for (int comp=0; comp<3; comp++) {
+            Efield_fp[lev][comp]->setVal(0);
+            Bfield_fp[lev][comp]->setVal(0);
+        }
+    }
+}
+
+void
+WarpX::ChargeDensityGridProcessing ()
+{
+    for (int lev = 0; lev <= max_level; lev++) {
+        ApplyFilterandSumBoundaryRho (lev, lev, *rho_fp[lev], 0, 1);
+    }
+#ifdef WARPX_DIM_RZ
+    for (int lev = 0; lev <= max_level; lev++) {
+        ApplyInverseVolumeScalingToChargeDensity(rho_fp[lev].get(), lev);
+    }
+#endif
+}
+
 void
 WarpX::ComputeSpaceChargeField (bool const reset_fields)
 {
+    // Reset all E and B fields to 0, before calculating space-charge fields
```

Comment:
```suggestion
    // Reset rho, and all E and B fields to 0, before calculating space-charge fields
```
Just a small change, but I think it's a good idea to update documentation to match changed functionality.